File: Array & String/Reverse Words in a String/ReverseWords.java
class Solution {
    public String reverseWords(String s) {
        String[] x = s.split(" +");
        StringBuilder sb = new StringBuilder();
        for (int i = x.length - 1; i >= 0; i--) {
            sb.append(x[i]);
            sb.append(" ");
        }
        return sb.toString().trim();
    }
}

File: Array & String/Reverse Words in a String/ReverseWords.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> words;
        istringstream iss(s);
        string word;
        
        while (iss >> word) {
            words.push_back(word);
        }

        ostringstream oss;
        for (int i = words.size() - 1; i >= 0; --i) {
            oss << words[i];
            if (i > 0) {
                oss << " ";
            }
        }
        
        return oss.str();
    }
};